server/index.ts

```diff
@@ -144,6 +145,28 @@ app
       next();
     });
     server.use('/api/v1', routes);
+    server.get('/deep-link', (req, res) => {
+      const url = req.query.url as string;
+      if (!url) {
+        res.status(400);
+        res.send('URL was not supplied.');
+        return;
+      }
+
+      const parsedUrl = new URL(url);
+      const deepLinkURL = parsedUrl.searchParams.get('v');
+
+      res.send(`
+        <script type="text/javascript">
+            window.onload = function() {
+                window.location = 'youtube://${deepLinkURL}';
+                window.setTimeout(function() {
+                    window.location = "${url}";
+                }, 25);
+            };
+        </script>
+      `);
+    });
```

Comment:
I'm not sure what the purpose of this endpoint is?

Reply:
this allows for deeplinking, without it it keeps a tab open on mobile : fixes #1576

Reply:
Okay sure I can understand why you want to do this but why not just handle this on the JS side in the frontend instead of doing it in the API?

Reply:
Also what happens if the user doesn't have the youtube app installed on their phone?

Reply:
It has to be a new window for this to work , if it YouTube isn’t on their phone it will go to the YouTube site

Reply:
I also don't understand why this isn't done in the frontend. 🤔

Reply:
I've tried this but it seems it has to be a new window for this to work. You could probably render this from a view but I ran into problems trying that as well (its been a few months so i can't recall all the particulars)

Reply:
I don't understand what the need for a new window has to do with this being handled in the frontend, though?  This doesn't belong in the API.

Reply:
we can move this into a deeplink.html and serve that instead. But you don't want to load the entire app if you don't have to.

Reply:
i found a way todo it via UI only , it may not work for all phones, i only tested on my iOS phone.